### PR TITLE
(Cherry Pick) Allow wpcom site address login for Jetpack app

### PR DESCRIPTION
**Pull request by ashiagr**
This PR cherry-picks an already reviewed and merged commit (see: https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/80 for details) so that it targets a correct release branch.

(cherry picked from commit https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/commit/246294ddc66acd22d7e92c2c1c8feddb4bb57d2e)

### CODE REVIEW

**Comment by AliSoftware** [comment]
> image

Not sure why CircleCI suddenly got re-enabled on that repo, leading to the `ci/circleci` check re-appearing on this PR — while we've [migrated this repo to Buidlkite a while ago now](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/74) 🤷  I've re-disabled CircleCI from this repo and removed the webhook so that it stops happening.

TL;DR: You can safely disregard the `ci/circleci: Build Error` check failure on that PR and merge it regardless of this check failure (as long as the other, Buildkite checks are green, of course).

**Review by zwarm — APPROVED** [review]
LGTM
